### PR TITLE
handle LoadBalancers (TargetGroups on AWS) removed

### PR DESCRIPTION
**Pull request by fabiokung**
When a LoadBalancer is removed outside of Titus, the reconciliation loop will (on each iteration):

1. Detect the load balancer gone, mark all its job associations as orphan.
2. Update orphan associations as Dissociated on the store.
3. Reap (delete) dissociated records when no more operations are pending.

### CODE REVIEW

File: titus-api/src/main/java/com/netflix/titus/api/connector/cloud/LoadBalancer.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    public State getState() {
+        return state;
+    }
+
+    public Set<String> getRegisteredIps() {
+        return registeredIps;
+    }
+
+    @Override
+    public String toString() {
```
Comment (tbak) [inline titus-api/src/main/java/com/netflix/titus/api/connector/cloud/LoadBalancer.java]: hash/equals for completeness?

File: titus-api/src/main/java/com/netflix/titus/api/connector/cloud/LoadBalancerConnector.java [hunk 2]
```diff
@@ -37,5 +37,5 @@
      */
     Completable isValid(String loadBalancerId);
 
-    Single<Set<String>> getRegisteredIps(String loadBalancerId);
+    Single<LoadBalancer> getLoadBalancer(String loadBalancerId);
```
Comment (tbak) [inline titus-api/src/main/java/com/netflix/titus/api/connector/cloud/LoadBalancerConnector.java]: Whenever we modify or create a new method with RX result, we should use Spring-Reactor, not RxJava.
If we do not, we will have to do it anyway in a few months from now.

Comment (tbak) [reply titus-api/src/main/java/com/netflix/titus/api/connector/cloud/LoadBalancerConnector.java]: I have created a little script to estimate RxJava vs Spring-Reactor usage in our code base: https://github.com/Netflix/titus-control-plane/blob/master/rx.sh

**Comment by fabiokung** [comment]
@gabrielhartmann @corindwyer @andrew-leung PTAL

**Pull request merged**